applications/minotari_console_wallet/src/automation/commands.rs

```diff
@@ -818,178 +916,184 @@ pub async fn command_runner(
+                println!("Concluded step 3 'faucet-encumber-aggregate-utxo'");
+                println!("Send '{}' to parties for step 4", get_file_name(STEP_3_PARTIES, None));
+                println!();
             },
-            FaucetCreateScriptSig(args) => {
-                let script = TariScript::from_hex(&args.input_script)
-                    .map_err(|e| CommandError::InvalidArgument(e.to_string()))?;
-                let input_data = ExecutionStack::from_hex(&args.input_stack)
-                    .map_err(|e| CommandError::InvalidArgument(e.to_string()))?;
-                let commitment =
-                    Commitment::from_hex(&args.commitment).map_err(|e| CommandError::InvalidArgument(e.to_string()))?;
-                let ephemeral_commitment = Commitment::from_hex(&args.ephemeral_commitment)
-                    .map_err(|e| CommandError::InvalidArgument(e.to_string()))?;
-                let ephemeral_pubkey = PublicKey::from(args.ephemeral_pubkey);
+            FaucetCreateInputOutputSigs(args) => {
+                // Read session info
+                let session_info = read_session_info(&args.session_id, None)?;
+                // Read leader input
+                let leader_info = read_and_verify::<Step3OutputsForParties>(
+                    &args.session_id,
+                    &get_file_name(STEP_3_PARTIES, None),
+                    &session_info,
+                )?;
+                // Read own party info
+                let party_info = read_and_verify::<Step2OutputsForSelf>(
+                    &args.session_id,
+                    &get_file_name(STEP_2_SELF, None),
+                    &session_info,
+                )?;
+
+                // Script signature
                 let challenge = TransactionInput::build_script_signature_challenge(
                     &TransactionInputVersion::get_current_version(),
-                    &ephemeral_commitment,
-                    &ephemeral_pubkey,
-                    &script,
-                    &input_data,
-                    &args.total_script_key.into(),
-                    &commitment,
+                    &leader_info.script_signature_ephemeral_commitment,
+                    &leader_info.script_signature_ephemeral_pubkey,
+                    &leader_info.input_script,
+                    &leader_info.input_stack,
+                    &leader_info.total_script_key,
+                    &Commitment::from_hex(&session_info.commitment_to_spend)?,
                 );
 
+                let mut script_signature = Signature::default();
                 match key_manager_service
-                    .sign_with_nonce_and_message(&args.private_key_id, &args.secret_nonce_key_id, &challenge)
+                    .sign_with_nonce_and_message(
+                        &party_info.wallet_spend_key_id,
+                        &party_info.script_nonce_key_id,
+                        &challenge,
+                    )
                     .await
```

Comment:
nit: this is fine as is, but we should be able to just exit here with `?` as there is no point in carrieing if there is a problem with the signature, same goes for below. Doing this we don't have to check if they are default below.

Reply:
I opted for this to give comprehensice feedback to the user.